src/core/connection.c

```diff
@@ -2190,20 +2195,27 @@ QuicConnRecvResumptionTicket(
             IndicateResumed,
             Connection,
             "Indicating QUIC_CONNECTION_EVENT_RESUMED");
-        ResumptionAccepted =
-            QUIC_SUCCEEDED(QuicConnIndicateEvent(Connection, &Event));
-
-        if (ResumptionAccepted) {
+        Status = QuicConnIndicateEvent(Connection, &Event);
+        Connection->Crypto.TicketValidationPending = Status == QUIC_STATUS_PENDING;
```

Comment:
This flag may have already been cleared in the callback. Don't set it to pending again if `Status == QUIC_STATUS_PENDING`. Same for cert validation path.